qiskit/primitives/base_estimator.py

```diff
@@ -372,6 +391,71 @@ def __call__(
             **run_options,
         )
 
+    def run(
+        self,
+        circuits: Sequence[int | QuantumCircuit],
+        observables: Sequence[int | SparsePauliOp],
+        parameter_values: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> Job:
+        """Run the job of the estimation of expectation value(s).
+
+        ``circuits``, ``observables``, and ``parameter_values`` should have the same
+        length. The i-th element of the result is the expectation of observable
+
+        .. code-block:: python
+
+            obs = self.observables[observables[i]]
+
+        for the state prepared by
+
+        .. code-block:: python
+
+            circ = self.circuits[circuits[i]]
+
+        with bound parameters
+
+        .. code-block:: python
+
+            values = parameter_values[i].
+
+        Args:
+            circuits: the list of circuit indices or circuit objects.
+            observables: the list of observable indices or observable objects.
+            parameter_values: concrete parameters to be bound.
+            run_options: runtime options used for circuit execution.
+
+        Returns:
+            The job object of EstimatorResult.
+        """
+        circuit_indices: list[int] = []
+        for circuit in circuits:
+            if isinstance(circuit, (int, np.integer)):
+                circuit_indices.append(circuit)
+            else:
+                try:
+                    circuit_indices.append(next(_finditer(id(circuit), self._circuit_ids)))
+                except StopIteration:
+                    self._append_circuit(circuit)
+                    circuit_indices.append(len(self._circuits) - 1)
+        observable_indices = []
+        for observable in observables:
+            if isinstance(observable, (int, np.integer)):
+                observable_indices.append(observable)
+            else:
+                try:
+                    observable_indices.append(next(_finditer(id(observable), self._observable_ids)))
+                except StopIteration:
+                    self._append_observable(observable)
+                    observable_indices.append(len(self._observables) - 1)
+
+        circuit_indices, observable_indices, parameter_values = self._validation(
+            circuit_indices, observable_indices, parameter_values
+        )
+        return self._submit(
+            lambda: self._call(circuit_indices, observable_indices, parameter_values, **run_options)
```

Comment:
I think the base class is a bit too complex. It's not obvious to me what `_call()` and `_submit()` are used for, even though they're both abstract methods. The child classes should have more freedom in how they want to implement `run()`.

Reply:
The reason I implemented them separately is we want to reuse `_call` in the first implementation. Type hint said their role but it may be complex... I will reconsider it to have more freedom.

Reply:
Simply I made `_run` and `_call` abstract methods. In the future, `_call` will be deprecated along with the original method `__call__`.

Reply:
Now it has simple `_run` methods in the base classes